src/components/FindMovie/FindMovie.tsx

```diff
@@ -1,54 +1,131 @@
-import React from 'react';
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import React, { useState } from 'react';
+import { moviesfromServer } from '../../public/api';
 import './FindMovie.scss';
 
-// import { MovieCard } from '../MovieCard';
-
-export const FindMovie: React.FC = () => (
-  <>
-    <form className="find-movie">
-      <div className="field">
-        <label className="label" htmlFor="movie-title">
-          Movie title
-        </label>
-
-        <div className="control">
-          <input
-            type="text"
-            id="movie-title"
-            placeholder="Enter a title to search"
-            className="input is-danger"
-          />
-        </div>
+import { MovieCard } from '../MovieCard';
 
-        <p className="help is-danger">
-          Can&apos;t find a movie with such a title
-        </p>
-      </div>
+interface Props {
+  setMovies: (a: Movie[]) => void
+  movies: Movie[];
+}
+
+export const FindMovie: React.FC <Props> = ({
+  setMovies,
+  movies = [],
+}) => {
+  const [input, setInput] = useState('');
+  const [movie, setMovie] = useState<Movie | null>(null);
+  const [error, setError] = useState(false);
+  const [errorEmpty, setErrorEmpty] = useState(false);
+  const [sameError, setSomeError] = useState(false);
+
+  const findMovies = async () => {
+    setSomeError(false);
+    if (input) {
+      const data = await moviesfromServer(input);
+
+      if (data.Response !== 'False') {
+        setMovie(await data);
+      } else {
+        setMovie(null);
+        setError(true);
+      }
+    } else {
+      setError(true);
+    }
+
+    setInput('');
+  };
+
+  const add = async () => {
+    if (movie) {
+      if (!movies.some(m => m.Title === movie.Title)) {
+        setMovies([...movies, movie]);
+      } else {
+        setSomeError(true);
+      }
+    }
 
-      <div className="field is-grouped">
-        <div className="control">
-          <button
-            type="button"
-            className="button is-light"
+    setMovie(null);
+  };
+
+  return (
+    <>
+      <form className="find-movie">
+        <div className="field">
+          <label
+            className="label"
+            htmlFor="movie-title"
           >
-            Find a movie
-          </button>
+            Movie title
+          </label>
+
+          <div className="control">
+            <input
+              value={input}
+              type="text"
+              id="movie-title"
+              placeholder="Enter a title to search"
+              className={error ? 'input is-danger' : 'input'}
+              onChange={(event) => {
+                setInput(event.target.value);
+                setError(false);
+                setErrorEmpty(false);
+                setSomeError(false);
+              }}
+            />
+          </div>
+
+          {!!(error) && (
```

Comment:
error already boolean (everywhere)
```suggestion
          {error && (
```